packages/core/src/utils.js

```diff
@@ -655,28 +656,63 @@ export function retrieveSchema(schema, rootSchema = {}, formData = {}) {
     return {};
   }
   let resolvedSchema = resolveSchema(schema, rootSchema, formData);
-  if ("allOf" in schema) {
-    try {
-      resolvedSchema = mergeAllOf({
```

Comment:
What's the reasoning behind dropping `json-schema-merge-allof` in favor of a custom solution?

Reply:
I don't think `json-schema-merge-allof` covers a lot of scenarios we need (such as allOf with if / then / else).

Also, the way `json-schema-merge-allof` gives errors is not great; it's hard for us to capture those errors and create react-jsonschema-form errors from them.

There's also this issue: https://github.com/rjsf-team/react-jsonschema-form/issues/1821

Reply:
That's fair, I just wonder if maybe that work should be split out into either another library all together, or another PR. As it is, with our large deployment of RJSF across all of our customer sites this PR scares the hell out of me.

Reply:
Yeah, I don't think this PR is ready yet. It needs more work, first to minimize changes so that only the bare minimum needed for if/then/else remains, and then to verify functionality using tests.